faucet/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use encointer_primitives::{
+	communities::CommunityIdentifier,
+	reputation_commitments::{DescriptorType, FromStr, PurposeIdType},
+	scheduler::CeremonyIndexType,
+};
+
+use core::marker::PhantomData;
+use frame_support::{
+	traits::{Currency, ExistenceRequirement::KeepAlive, Get},
+	PalletId,
+};
+use frame_system::{self as frame_system, ensure_signed};
+use log::info;
+use sp_runtime::{traits::AccountIdConversion, Saturating};
+use sp_std::convert::TryInto;
+
+// Logger target
+const LOG: &str = "encointer";
+
+pub use pallet::*;
+
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + encointer_reputation_commitments::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type Currency: Currency<Self::AccountId>;
+		type FaucetMaster: EnsureOrigin<Self::RuntimeOrigin>;
+
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Transfer some balance to another account.
+		#[pallet::call_index(0)]
+		#[pallet::weight(10_000)]
+		pub fn drip(
+			origin: OriginFor<T>,
+			cid: CommunityIdentifier,
+			cindex: CeremonyIndexType,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+
+			<encointer_reputation_commitments::Pallet<T>>::do_commit_reputation(
+				&from,
+				cid,
+				cindex,
+				Self::reputation_commitments_purpose_id(),
+				None,
+			)?;
+
+			T::Currency::transfer(&Self::account_id(), &from, Self::drip_amount(), KeepAlive)
+				.map_err(|_| <Error<T>>::FaucetEmpty)?;
+
+			Self::deposit_event(Event::Dripped(from, Self::drip_amount()));
+
+			Ok(().into())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(10_000)]
+		pub fn set_drip_amount(
+			origin: OriginFor<T>,
+			drip_amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			T::FaucetMaster::ensure_origin(origin)?;
+			<DripAmount<T>>::put(drip_amount);
+			info!(target: LOG, "set drip amount to {:?}", drip_amount);
+			Self::deposit_event(Event::DripAmountUpdated(drip_amount));
+			Ok(().into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// The account ID of the faucet.
+		///
+		/// This actually does computation. If you need to keep using it, then make sure you cache the
+		/// value and only call this once.
+		pub fn account_id() -> T::AccountId {
+			T::PalletId::get().into_account_truncating()
+		}
+
```

Comment:
fine for now. Just a comment: looking ahead, I'd prefer if this would be called `pot_account(Option<FaucetIndex>)` and the account should be distinct per faucet instance - as there may be multiple faucets with different sponsors and dripping amounts or whitelisted cids for the same token